Mathlib/Analysis/Asymptotics/TVS.lean

```diff
@@ -113,75 +160,151 @@ theorem IsLittleOTVS.congr (h : f₁ =o[𝕜; l] g₁) (hf : ∀ x, f₁ x = f
   h.congr' (univ_mem' hf) (univ_mem' hg)
 
 theorem IsLittleOTVS.congr_left (h : f₁ =o[𝕜; l] g) (hf : ∀ x, f₁ x = f₂ x) : f₂ =o[𝕜; l] g :=
-  h.congr hf fun _ => rfl
+  h.congr hf fun _ ↦ rfl
 
 theorem IsLittleOTVS.congr_right (h : f =o[𝕜; l] g₁) (hg : ∀ x, g₁ x = g₂ x) : f =o[𝕜; l] g₂ :=
-  h.congr (fun _ => rfl) hg
+  h.congr (fun _ ↦ rfl) hg
 
 end congr
 
 variable {l l₁ l₂ : Filter α} {f : α → E} {g : α → F}
 
-@[trans]
-theorem IsLittleOTVS.trans {k : α → G} (hfg : f =o[𝕜; l] g) (hgk : g =o[𝕜; l] k) :
+theorem IsLittleOTVS.isBigOTVS (h : f =o[𝕜; l] g) : f =O[𝕜; l] g := by
+  refine ⟨fun U hU ↦ ?_⟩
+  rcases h.1 U hU with ⟨V, hV₀, hV⟩
+  use V, hV₀
+  simpa using hV 1 one_ne_zero
+
+theorem IsBigOTVS.trans {k : α → G} (hfg : f =O[𝕜; l] g) (hgk : g =O[𝕜; l] k) : f =O[𝕜; l] k := by
+  refine ⟨fun U hU₀ ↦ ?_⟩
+  obtain ⟨V, hV₀, hV⟩ := hfg.1 U hU₀
+  obtain ⟨W, hW₀, hW⟩ := hgk.1 V hV₀
+  refine ⟨W, hW₀, ?_⟩
+  filter_upwards [hV, hW] with x hx₁ hx₂ using hx₁.trans hx₂
+
+instance instTransIsBigOTVSIsBigOTVS :
```

Comment:
Why do you register this instance by hand instead of putting `@[trans]` on the previous theorem?

Reply:
I tested, and `@[trans]` doesn't add a `Trans` instance. Also, I unfolded notation, because AFAIR, Lean doesn't alpha-reduce in these cases, which leads to worse discrimination tree keys.